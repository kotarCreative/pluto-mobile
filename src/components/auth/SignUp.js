/**
 * Sign up screen
 *
 * @format
 * @flow
 */

import React, { PureComponent } from 'react'
import { Image, StyleSheet, Text, TouchableHighlight, View } from 'react-native'
import { Actions } from 'react-native-router-flux'
import { getBottomSpace } from 'react-native-iphone-x-helper'

// Components
import { Button } from '@/components/shared/Button'
import { FormLayout } from '@/components/shared/FormLayout'
import { TextInput } from '@/components/shared/TextInput'
import { ViewWithBg } from '@/components/shared/ViewWithBg'

import { BRAND_SECONDARY_HIGHLIGHT, WHITE } from '@/constants/colors'
import logo from '@/assets/logos/wordmark.png'

// Styles
const styles = StyleSheet.create({
  accountBtn: {
    backgroundColor: BRAND_SECONDARY_HIGHLIGHT,
    padding: 10,
    paddingBottom: getBottomSpace() - 10
  },
  centeredContainer: {
    alignSelf: 'center',
    width: 200
  },
  footer: {
    right: 0,
    left: 0,
    position: 'absolute',
    bottom: 0
  },
  logo: {
    alignSelf: 'center',
    marginBottom: 60
  },
  loginBtnTxt: {
    textAlign: 'center',
    color: WHITE
  },
  signUpButton: {
    marginTop: 30,
    width: '100%'
  }
})

export class SignUp extends PureComponent {
  render() {
    return (
      <ViewWithBg>
        <View style={styles.centeredContainer}>
          <Image style={styles.logo} source={logo} />
          <FormLayout>
            <TextInput placeholder="Email"
                       onChangeText={text => {}}
                       />
            <TextInput placeholder="Password"
                       textContentType="password"
                       secureTextEntry
                       onChangeText={text => {}}
                       />
             <TextInput placeholder="Confirm Password"
                        textContentType="password"
                        secureTextEntry
                        onChangeText={text => {}}
                        />
          </FormLayout>
          <Button title="Sign Me Up"
                  style={styles.signUpButton}
                  onPress={() =>  "Clicked"}
                  disabled={true}
                  secondary={true}
                  />
        </View>
        <View style={styles.footer}>
          <TouchableHighlight
            onPress={() => Actions.pop()}
            style={styles.accountBtn}
            underlayColor={BRAND_SECONDARY_HIGHLIGHT}
          >
            <Text style={styles.loginBtnTxt}>Already have an account?</Text>
          </TouchableHighlight>
        </View>
      </ViewWithBg>
    )
  }
}
