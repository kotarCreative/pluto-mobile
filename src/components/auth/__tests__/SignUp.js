import React from 'react'
import renderer from 'react-test-renderer'

import { SignUp } from '../SignUp'


it('renders correctly with defaults', () => {
  const signup = renderer
    .create(<SignUp />)
    .toJSON()
  expect(signup).toMatchSnapshot()
});
