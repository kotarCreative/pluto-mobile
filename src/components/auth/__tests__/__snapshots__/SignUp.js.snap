// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": "100%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "padding": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "width": 200,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 60,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#FFFFFF",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Email"
            placeholderTextColor="#F5FCFF"
            selectionColor="#F5FCFF"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#FFFFFF",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#F5FCFF"
            secureTextEntry={true}
            selectionColor="#F5FCFF"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="Confirm Password"
            placeholderTextColor="#F5FCFF"
            secureTextEntry={true}
            selectionColor="#F5FCFF"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "padding": 5,
                "textAlign": "center",
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginTop": 30,
            "padding": 5,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "textAlign": "center",
            }
          }
        >
          Sign Me Up
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E38C3B",
            "padding": 10,
            "paddingBottom": -10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "textAlign": "center",
            }
          }
        >
          Already have an account?
        </Text>
      </View>
    </View>
  </View>
</View>
`;
